SQLite3/mORMot.pas

```diff
@@ -52244,7 +52244,9 @@ procedure TJSONSerializer.WriteObject(Value: TObject; Options: TTextWriterWriteO
               Add('"');
             AddDateTime(D64);
             if woDateTimeWithZSuffix in Options then
-              Add('Z');
+              if frac(D64)=0 then // FireFox can't decode short form "2017-01-01Z"

```

Comment:
won't it break some existing code?
perhaps worth a new option

Reply:
I don't think it's break something - in case of SynCommon - based clients deserialization will work with new format. In case with JavaScript  based client this is mandatory changes